### ISSUE TITLE
Operator '<' cannot be applied to types 'boolean' and 'never'.

### ISSUE THREAD

**Issue by na7ure-a** [microsoft/TypeScript#62149]
### 🔎 Search Terms

Operator '<' cannot be applied to types 'boolean' and 'never'.

### 🕗 Version & Regression Information

- This changed between versions 4.9.5 and 5.0.4

### ⏯ Playground Link

https://www.typescriptlang.org/zh/play/?target=7&ts=5.0.4#code/DYUwLgBAhhC8EDMrAM4gLAChSQEZ0WTSwHoSIBRAJyoHsqIqQAHesASwDsBzCFdsAFcoHWpyw5GIFAXwAyORAAUMADwRcASlLkA8mAAWIBp3oBbZBADG7KlcFmUYKJyvSs+eGCqCMmJjLw8ooqEOpaWAEESKggEArKahqaQA

### 💻 Code

```ts
let a = false
let b = false
// Error reporting situation
let res = b && (a < b)
// Other normal circumstances
b = true
res = b && (a < b)
res = false && (a < b)
```

### 🙁 Actual behavior

There were no errors before version 4.9.5
An error occurred after version 5.0.4

### 🙂 Expected behavior

This error should not be reported

**Comment by MartinJohns** [comment]
This is working as intended.

See also: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-5-0.html#forbidden-implicit-coercions-in-relational-operators

**Comment by na7ure-a** [comment]
> This is working as intended.
> 
> See also: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-5-0.html#forbidden-implicit-coercions-in-relational-operators

Thank you for your prompt handling, but why does an error only occur when b=false in the code?

**Comment by RyanCavanaugh** [comment]
This is basically TypeScript's way of telling you that the right operand of `&&` is not reachable
